src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,73 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  countOfTodos: number;
+  countOfCompletedTodos: number;
+  isInputDisabled: boolean;
+  fetchAddTodo: (title: string) => Promise<boolean>;
+  setErrorMessage: (error: Errors) => void;
+  UpdateAllTodosCompleted: () => void;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  countOfTodos,
+  countOfCompletedTodos,
+  isInputDisabled,
+  fetchAddTodo,
+  setErrorMessage,
+  UpdateAllTodosCompleted,
+}) => {
+  const [inputText, setInputText] = useState('');
+  const isAllTodosCompleted = countOfCompletedTodos === countOfTodos;
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const formattedTitle = inputText.trim();
+
+    if (formattedTitle) {
+      if (await fetchAddTodo(formattedTitle)) {
+        setInputText('');
+      }
```

Comment:
The function is well-structured, but consider adding a try-catch block for error handling during the fetch call.